manifests/init.pp

```diff
@@ -81,24 +85,47 @@
     target => $log_path,
   }
 
-  wget::fetch { 'selenium-server-standalone':
-    source             => $jar_url,
-    destination        => "${jar_path}/${jar_name}",
-    timeout            => $download_timeout,
-    nocheckcertificate => $nocheckcertificate,
-    execuser           => $user,
-    require            => File[$jar_path],
+  if $package != undef {
+
+    package {$package:
+      ensure => installed,
+    }
+
+  } else {
+
+    if (!defined(Package['wget'])) {
+      package {'wget':
+        ensure => installed,
+      }
+    }
+
+    if $nocheckcertificate {
+      $_nocheckcertificate = '--no-check-certificate'
+    } else {
+      $_nocheckcertificate = ''
+    }
+
+    exec {'fetch selenium-server-standalone':
+      path    => '/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin',
+      user    => $user,
+      command => "wget -T ${download_timeout} ${_nocheckcertificate} -O ${jar_path}/${jar_name} ${jar_url}",
+      creates => "${jar_path}/${jar_name}",
+      require => [File[$jar_path], Package['wget']],
+    }
```

Comment:
I'll be honest, this is going to be hard sell for me. Why write unit tests for this complicated exec when this is already handled by the `wget` module?